frontend/projects/upgrade/src/app/core/stratification-factors/store/stratification-factors.effects.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { fakeAsync, tick } from '@angular/core/testing';
+import { ActionsSubject } from '@ngrx/store';
+import { BehaviorSubject, of, throwError } from 'rxjs';
+import { StratificationFactorsEffects } from './stratification-factors.effects';
+import * as StratificationFactorsActions from './stratification-factors.actions';
+import { selectAllStratificationFactors } from './stratification-factors.selectors';
+import { StratificationFactor } from './stratification-factors.model';
+
+describe('StratificationFactorsEffects', () => {
+  let actions$: ActionsSubject;
+  let store$: any;
+  let stratificationFactorsDataService: any;
+  let router: any;
+  let service: StratificationFactorsEffects;
+  const mockData: StratificationFactor = { factor: 'favourite_food', values: { pizza: 10, burger: 5 } };
+
+  beforeEach(() => {
+    actions$ = new ActionsSubject();
+    store$ = new BehaviorSubject({});
+    store$.dispatch = jest.fn();
+    stratificationFactorsDataService = {};
+    router = {
+      navigate: jest.fn(),
+    };
+
+    service = new StratificationFactorsEffects(actions$, router, store$, stratificationFactorsDataService);
+  });
+
+  describe('fetchStratificationFactors$', () => {
+    it('should dispatch actionFetchStratificationFactorsSuccess on API call success', fakeAsync(() => {
+      stratificationFactorsDataService.fetchStratificationFactors = jest.fn().mockReturnValue(of(mockData[0]));
+      selectAllStratificationFactors.setResult(mockData[0]);
```

Comment:
mockData is an Object and mackData[0] will be undefined. Is this correct?

Reply:
Yes, you are correct. Resolved this in the latest commit.